lib/SlackBot.js

```diff
@@ -118,6 +118,73 @@ function Slackbot(configuration) {
             'webhooks at: http://MY_HOST:' + slack_botkit.config.port + '/slack/receive');
         webserver.post('/slack/receive', function(req, res) {
 
+            // is this an events api url handshake?
+            if (req.body.type === 'url_verification') {
+                slack_botkit.debug('Received url handshake');
+                res.status(200).json({
+                    challenge: req.body.challenge
+                });
+
+            }
+
+            // Receive messages and trigger events from the Events API
+            if (req.body.type === 'event_callback') {
+                // respond to events api with a 200
+                res.sendStatus(200);
+                if (req.body.event.user === req.body.authed_users[0]) {
+                    slack_botkit.debug('Got event from this bot user, ignoring it');
+                    return;
+                }
+
+                var message = {};
+                for (var key in req.body.event) {
+                    message[key] = req.body.event[key];
+                }
+                // let's normalize some of these fields to match the rtm message format
+                message.team = req.body.team_id;
+                message.events_api = true;
+                message.authed_users = req.body.authed_users;
+                slack_botkit.findTeamById(message.team, function(err, team) {
+                    if (err || !team) {
+                        slack_botkit.log.error('Received Events API message, but could not load team');
+                        if (err) {
+                            slack_botkit.log.error(err);
```

Comment:
if you return here, then you save yourself the if and keeps the indentation flatter.